### ISSUE TITLE
Deleted element not properly accounts for in creation of initial model.

### ISSUE THREAD

**Issue by amgrubb** [amgrubb/BloomingLeaf#166]
If I start creating a model and delete an element and keep creating a model I get the following error when I click the analysis button:

![screen shot 2018-10-16 at 9 23 59 pm](https://user-images.githubusercontent.com/6596937/47056338-db040a80-d189-11e8-8dec-ba825e36ab81.png)

There error appears to be from the following code.
onFunction.js Lines 38-66:
```
function reassignIntentionIDs() {
	var elements = graph.getElements();
	var intentions = model.intentions;
 	var currID = 0;
	var currIDStr;
	for (var i = 0; i < intentions.length; i++) {
		var intention = intentions[i];
 		if (parseInt(intention.nodeID) !== currID) {
 			// The current intention's ID must be reassigned
		
			// Find the intention's cell
			var cell;
			for (var j = 0; j < elements.length; j++) {
				if (elements[i].attributes.nodeID === intention.nodeID) {
					cell = elements[i];
				}
			}
 			currIDStr = currID.toString();
 			while (currIDStr.length < 4){
	                currIDStr = '0' + currIDStr;
	        }
			cell.attributes.nodeID = currIDStr;
			intention.setNewID(currIDStr);
		}
 		currID += 1;
	}
 	Intention.numOfCreatedInstances = currID;
	Link.numOfCreatedInstances = currID;
}
```

**Comment by oibrahim22** [comment]
Lucy and I tried to recreate the error and we couldn't. We believe that Lucy's solution to issue #176 took care of this one too.

**Comment by amgrubb** [comment]
May be fixed by #180
